### PR TITLE
Explain --fetch-external; update some other stuff

### PR DESCRIPTION
**Pull request by haltman-at**
Adds documentation of `--fetch-external`.  Also expanded on breakpoints while I was at it.

### CODE REVIEW

File: src/docs/truffle/getting-started/debugging-your-contracts.md [hunk 1]
```diff
@@ -96,6 +104,18 @@ $ truffle debug
 
 Regardless of how you start the debugger, once it is running you are not limited to debugging only the transaction you launched it with; it is possible to unload the current transaction and load a new one, as described below.
 
+You can specify the network you want to debug on with the `--network` option:
+
+```shell
+$ truffle debug <transaction hash> --network <network>
+```
+
+And with the `--fetch-external` option ([see below](#debugging-external-contracts-with-verified-source)), you can debug contract instances outside your project that have verified source code on Etherscan.  (Support for retrieving code from Sourcify as well is planned.)  When using this option, you must specify a transaction hash to debug, and you will not be able to switch transactions from inside the debugger.
```
Comment (gnidan) [inline src/docs/truffle/getting-started/debugging-your-contracts.md:113]: Probably needs link to Sourcify (and for consistency, also Etherscan)

File: src/docs/truffle/getting-started/debugging-your-contracts.md [hunk 2]
```diff
@@ -107,6 +127,21 @@ however, the debugger can likely avoid the initial recompile, speeding up
...
+This option can also be abbreviated `-x`.
+
+If you have an Etherscan API key, you can include it in your configuration file and the debugger will use it when downloading source from Etherscan.  Including this can speed up downloads.
+
+Example:
+```javascript
+etherscan: {
+  apiKey: "0123456789abcdef0123456789abcdef" //replace this with your API key if you have one
+}
+```
```
Comment (gnidan) [inline src/docs/truffle/getting-started/debugging-your-contracts.md:143]: Let's wrap this with contextual info:

```suggestion
```javascript
module.exports = {
  /* ... rest of truffle-config.js ... */
  etherscan: {
    apiKey: "0123456789abcdef0123456789abcdef" //replace this with your API key if you have one
  }
}
```
```

File: src/docs/truffle/getting-started/debugging-your-contracts.md (hunk 1 above)
Comment (haltman-at) [reply src/docs/truffle/getting-started/debugging-your-contracts.md:113]: Oh, good point.

File: src/docs/truffle/getting-started/debugging-your-contracts.md (hunk 2 above)
Comment (haltman-at) [reply src/docs/truffle/getting-started/debugging-your-contracts.md:143]: Good idea!

**Pull request merged**